ios/Classes/TiBarcodeModule.m

```diff
@@ -0,0 +1,564 @@
+  [[barcodeViewController scanner] setCamera:selectedCamera ?: MTBCameraBack error:&cameraError];
+
+  if (displayedMessage != nil) {
+    [[barcodeViewController overlayView] setDisplayMessage:displayedMessage];
+  }
+  if (cameraError) {
+    [self fireEvent:@"error"
+         withObject:@{
+           @"message" : [cameraError localizedDescription] ?: @"Unknown error occurred."
+         }];
+  }
+
+  [[barcodeViewController scanner] setTorchMode:MTBTorchModeOff];
+
+  [[barcodeViewController scanner] startScanningWithResultBlock:^(NSArray *codes) {
+    if (!codes || [codes count] == 0) {
+      return;
+    }
+    [self handleSuccessResult:[(AVMetadataMachineReadableCodeObject *)[codes firstObject] stringValue]];
+
+    if (!keepOpen) {
+      [self closeScanner];
+    }
+  }
+                                                          error:&error];
+
+  if (error) {
+    [self fireEvent:@"error"
+         withObject:@{
+           @"message" : [error localizedDescription] ?: @"Unknown error occurred."
+         }];
+
+    if (!keepOpen) {
+      [self closeScanner];
+    }
+  }
+
+  [[[[TiApp app] controller] topPresentedController] presentViewController:barcodeViewController
+                                                                  animated:animate
+                                                                completion:^{
+                                                                  [[barcodeViewController scanner] setTorchMode:selectedLEDMode ?: MTBTorchModeOff];
+                                                                }];
+}
+
+- (void)freezeCapture:(id)unused
+{
+  ENSURE_UI_THREAD(freezeCapture, unused);
+  [[barcodeViewController scanner] freezeCapture];
+}
+
+- (void)unfreezeCapture:(id)unused
+{
+  ENSURE_UI_THREAD(unfreezeCapture, unused);
+  [[barcodeViewController scanner] unfreezeCapture];
+}
+
+- (void)captureStillImage:(id)value
+{
+  ENSURE_UI_THREAD(captureStillImage, value);
+  ENSURE_SINGLE_ARG(value, KrollCallback);
+
+  [[barcodeViewController scanner] captureStillImage:^(UIImage *image, NSError *error) {
+    TiBlob *blob = [[TiBlob alloc] _initWithPageContext:[self pageContext]];
+    [blob setImage:image];
+    [blob setMimeType:@"image/png" type:TiBlobTypeImage];
+
+    NSDictionary *event = [NSDictionary dictionaryWithObject:blob forKey:@"image"];
+    [self _fireEventToListener:@"blob" withObject:event listener:(KrollCallback *)value thisObject:nil];
+  }];
+}
+
+- (void)cancel:(id)unused
+{
+  ENSURE_UI_THREAD(cancel, unused);
+
+  [self closeScanner];
+  [self fireEvent:@"cancel" withObject:nil];
+}
+
+- (void)setUseLED:(id)value
+{
+  ENSURE_TYPE(value, NSNumber);
+  [self replaceValue:value forKey:@"useLED" notification:NO];
+
+  selectedLEDMode = [TiUtils boolValue:value def:YES] ? MTBTorchModeOn : MTBTorchModeOff;
+
+  if (barcodeViewController) {
```

Comment:
Please always compare against nil, since `barcodeViewController ` is not a boolean.